### PR TITLE
Pandamonium_main.lua

### PR DESCRIPTION
**Pull request by Sparks1992**
Forum Link: http://forums.heroesofnewerth.com/showthread.php?577675-Pandamonium-Bot

### CODE REVIEW

**Comment by Anakonda** [comment]
Please include bot file in the PR

**Comment by Sparks1992** [comment]
Like i did with Midas bot? or in the same pull request?

**Comment by Anakonda** [comment]
In the same pull request.

**Comment by Sparks1992** [comment]
Could you guide me a little please? First time i'm using GitHub :) how do i add another file in the same pull request? Attach with drag & drop in the comment?

**Comment by Anakonda** [comment]
I think you should try this: https://desktop.github.com/

File: Pandamonium_main.lua [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+		if nDistSq > 800 * 800 then
+			if nLastHarassUtility > behaviorLib.diveThreshold or core.NumberElements(core.GetTowersThreateningPosition(vecTargetPosition, nMyExtraRange, core.myTeam)) == 0 then
+				local _, sortedTable = HoN.GetUnitsInRadius(vecTargetPosition, 1000, core.UNIT_MASK_HERO + core.UNIT_MASK_ALIVE, true)
+				local EnemyHeroes = sortedTable.EnemyHeroes
+				if core.NumberElements(EnemyHeroes) == 1 then
+					bActionTaken = core.OrderItemPosition(botBrain, unitSelf, itemPortalKey, vecTargetPosition)
+				end
+			end
+		end
+	end
```
Comment (Anakonda) [inline Pandamonium_main.lua]: This seems like extra end, and I'm getting syntax error.

**Comment by Anakonda** [comment]
Please check indentation.

**Comment by Sparks1992** [comment]
should be ok now...i don't think there was any problems with the if-end syntax...but one skill was named wrong xD

File: Pandamonium_main.lua [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+	local bActionTaken = false
+	local nMinimumCreeps = 3
+	local unitSelf = core.unitSelf
+	local abilFlurry = skills.abilFlurry
+	
+	if abilFlurry:CanActivate() and core.unitSelf:GetManaPercent() > 0.60 then
+		local tCreeps = core.localUnits["EnemyCreeps"]
+		local nNumberCreeps =  core.NumberElements(tCreeps)
+		local vecTarget = core.GetGroupCenter(tCreeps)
+		local nDistSq = Vector3.Distance2DSq(unitSelf:GetPosition(), vecTarget)
```
Comment (Anakonda) [inline Pandamonium_main.lua]: bad argument 2 to Distance2DSq (Vector3 expected, got nil)

**Comment by Sparks1992** [comment]
Removed the error, should be ok now

**Pull request merged**